crypto/bio/b_sock2.c

```diff
@@ -115,6 +115,58 @@ int BIO_connect(int sock, const BIO_ADDR *addr, int options)
     return 1;
 }
 
+/*-
+ * BIO_localaddr - Sets local address for socket
+ * @sock: the socket to set
+ * @addr: local address to bind to
+ * @options: BIO socket options
+ *
+ * Binds to the address using the given socket and options, then
+ * starts listening for incoming connections.
+ *
+ * Options can be a combination of the following:
+ * - BIO_SOCK_REUSEADDR: Try to reuse the address and port combination
+ *   for a recently closed port.
+ *
+ * When restarting the program it could be that the port is still in use.  If
+ * you set to BIO_SOCK_REUSEADDR option it will try to reuse the port anyway.
+ * It's recommended that you use this.
+ */
+
+int BIO_localaddr(int sock, const BIO_ADDR *addr, int options)
+{
+    int on = 1;
+    int socktype;
+
+    if (sock == -1) {
```

Comment:
I am not sure we should do this.

Reply:
It's copied from the existing BIO_listen.